src/java.net.http/share/classes/java/net/http/HttpRequest.java

```diff
@@ -303,6 +305,75 @@ protected HttpRequest() {}
         return new HttpRequestBuilderImpl(uri);
     }
 
+    /**
+     * Creates a {@code Builder} seeded from an {@code HttpRequest}.
+     *
+     * This method returns a {@code Builder} whose state is copied from the
+     * given request, subject to the given filter. This builder can be used to
+     * build an {@code HttpRequest}, equivalent to the original, while allowing
+     * amendment of the request state prior to construction - for example,
+     * adding additional headers.
+     *
+     * <p> The {@code filter} is applied to each header name value pair as they
+     * are copied from the given request. When completed, only headers that
+     * satisfy the condition as laid out by the {@code filter} will be present
+     * in the {@code Builder} returned from this method.
+     *
+     * @apiNote
+     * The following scenarios demonstrate typical use-cases of the filter.
+     * Given an {@code HttpRequest} <em>request</em>:
+     * <br><br>
+     * <ul>
+     *  <li> Retain all headers:
+     *  <pre>{@code HttpRequest.newBuilder(request, (n, v) -> true)}</pre>
+     *
+     *  <li> Remove all headers:
+     *  <pre>{@code HttpRequest.newBuilder(request, (n, v) -> false)}</pre>
+     *
+     *  <li> Remove a particular header (e.g. Foo-Bar):
+     *  <pre>{@code HttpRequest.newBuilder(request, (name, value) -> name.equalsIgnoreCase("Foo-Bar"))}</pre>
+     * </ul>
```

Comment:
Does the example above not result in "Foo-Bar" being the only header being copied across? So, maybe it should say "Retain a particular header"

Reply:
Ah! Good catch - I believe the code is buggy. It should have been `!name.equalsIgnoreCase(...)`.

Reply:
Patrick - can you add a test case with exactly **that** use case? I know you have similar tests which remove `"testName1"`, but it would be good to have a test that verifies that **the exact code** shown in the `@apiNote` works as expected. Your tests should have caught that.

Reply:
Well spotted, Michael. I've fixed that typo in the example, and added a test case to cover it. See commit https://github.com/openjdk/jdk/pull/1059/commits/68ea89e1c55735adc9230b8abc07e1df3ffc03ca